### ISSUE TITLE
prep_batch_tops.csh

### ISSUE THREAD

**Issue by khakim** [gmtsar/gmtsar#48]
Dear all,

I have a problem with prep_batch_tops.csh, getting errors as below

preproc_batch_tops.csh data.in dem.grd 2
grdfilter (api_alloc_grid): Error: Could not reallocate memory [33527.74 Gb, 9000034800016 items of 4 bytes]
grd2xyz: File flt.grd not found
[Session gmt (0)]: Error returned from GMT API: GMT_FILE_NOT_FOUND (16)
rm: No match.

How do I fix the errors.

I appreciate for helps.

**Comment by Xiaohua-Eric-Xu** [comment]
Not sure how your DEM could take 3E5 Gb memory. Maybe cut to your image area before use? 
Or could you do a gmt grdinfo on your dem.grd

**Comment by khakim** [comment]
Thank very much for your help.  
You are right that the problem comes from the dem.
After I cheched the dem using grdinfo, the dem was already modified when I processed SBAS in previous processing.
As the results of SBAS is bad (deformation signal disappeared, it may be considered as atmospheric phase screen), I do reprocessing from creating "raw" data using previous "raw_orig" and "topo" folders. 
Now, I am trying a smaller images for SBAS, hopefully resulting the better results.

Below grid info the dem when running "prep_batch-tops.csh"
dem.grd: Title: Digital Elevation Model Error
dem.grd: Command: sbas intf.tab scene.tab 169 46 3125 2000 -smooth 5 -atm 3 -wavelength 0.0554658 -incidence 37 -range 851942.094342839 -rms -dem 
dem.grd: Remark: DEM error estimated from SBAS (m)
dem.grd: Pixel node registration used [Cartesian grid]
dem.grd: Grid file format: nf = GMT netCDF format (32-bit float), COARDS, CF-1.5
dem.grd: x_min: 0 x_max: 25000 x_inc: 8 name: x n_columns: 3125
dem.grd: y_min: 0 y_max: 4000 y_inc: 2 name: y n_rows: 2000
dem.grd: z_min: -134.52532959 z_max: 58.501625061 name: z
dem.grd: scale_factor: 1 add_offset: 0
dem.grd: format: netCDF-4 chunk_size: 131,134 shuffle: on deflation_level: 3
